src/js/components/sharedComponents/InfoContent.tsx

```diff
@@ -1,33 +1,136 @@
 import React, { FunctionComponent, useEffect, useState } from 'react';
 import { useSelector } from 'react-redux';
+import convert from 'xml-js';
 
 import { RootState } from 'js/store';
 
 import { downloadData } from '../../../../configs/modal.config';
 
+interface MetadataContent {
+  title?: string;
+  functionOrPurpose?: string;
+  geographicCoverage?: string;
+  tags?: string;
+  overview?: string;
+}
+
+interface ContentError {
+  error: string;
+}
+
+type Content = MetadataContent | ContentError;
+type SharingHost = string | undefined;
```

Comment:
instead of declaring this here, can we do the following: 1) in `appSettings/types` lets change sharinghost from optional parameter to required 2) can we add default sharinghost in appSettings/reducers to be 'https://www.arcgis.com'. This way we can read it directly from appSettings and we do not need to declare type or potentially have it as undefined because it will always be defined in that default reducer.